server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -874,6 +874,13 @@ public static String cleanTruncate(String s, int length) {
         return s.substring(0, length);
     }
 
+    public static String requireNonEmpty(String str, String message) {
```

Comment:
Can you add a javadocs for this method to indicate what it does and what is thrown?